src/main/scala/org/broadinstitute/hail/expr/AST.scala

```diff
@@ -1631,6 +1631,191 @@ case class BinaryOp(posn: Position, lhs: AST, operation: String, rhs: AST) exten
       s.r.findFirstIn(t).isDefined
     }
 
+    case ("+", TArray(elementType)) =>
+      ((lhs.`type`, rhs.`type`, elementType): @unchecked) match {
+        case (t: TNumeric, TArray(_), TDouble) => AST.evalCompose[Any, IndexedSeq[_]](ec, lhs, rhs) { case (num, arr) =>
+          val d = DoubleNumericConversion.to(num)
+          arr.map(x => DoubleNumericConversion.to(x) + d)
+        }
```

Comment:
I suspect we can abstract a bit more here using these functions:

```scala
  def liftArrayOpToEval[T](ec: EvalContext, lhs: AST, rhs: AST, coerce: Any => T, binOp: (T, T) => T): () => Any =
    liftEval[Any, IndexedSeq[Any]](ec, lhs, rhs, liftArrayOp(coerce, binOp))

  def liftArrayOp[T](coerce: Any => T, binOp: (T, T) => T): (Any, IndexedSeq[Any]) => IndexedSeq[T] =
    { case (num, arr) => arr.map(x => binOp(coerce(x), coerce(num))) }

  def liftEval[T1, T2](ec: EvalContext, lhs: AST, rhs: AST, g: (T1, T2) => Any): () => Any =
    AST.evalCompose[Any, IndexedSeq[_]](ec, lhs, rhs)(g)
```

which should permit defining this case as:

```scala
case (t: TNumeric, TArray(_), TDouble) => liftArrayOpToEval[Double](ec, lhs, rhs, DoubleNumericConversion.to, { _ + _})
```